--- a/core.c
+++ b/core.c
@@ -450,7 +450,6 @@ static void mwl_set_ht_caps(struct mwl_priv *priv,
 	}
 
 	ieee80211_hw_set(hw, AMPDU_AGGREGATION);
-	ieee80211_hw_set(hw, SUPPORTS_AMSDU_IN_AMPDU);
 	band->ht_cap.ampdu_factor = IEEE80211_HT_MAX_AMPDU_64K;
 	band->ht_cap.ampdu_density = IEEE80211_HT_MPDU_DENSITY_4;
 
@@ -982,7 +981,7 @@ struct ieee80211_hw *mwl_alloc_hw(int bus_type,
 	priv->use_short_preamble = false;
 	priv->disable_2g = false;
 	priv->disable_5g = false;
-	priv->tx_amsdu = true;
+	priv->tx_amsdu = false;
 	priv->hif.bus = bus_type;
 	priv->hif.ops = ops;
 	priv->hif.priv = (char *)priv + ALIGN(sizeof(*priv), NETDEV_ALIGN);
